configure.ac

```diff
@@ -1324,47 +1319,8 @@ dnl ***************************************************************************
 dnl rabbitmq-c headers/libraries
 dnl ***************************************************************************
 
+PKG_CHECK_MODULES(LIBRABBITMQ, librabbitmq >= $LIBRABBITMQ_MIN_VERSION, with_librabbitmq_client="yes", with_librabbitmq_client="no")
 
-LIBRABBITMQ_INTERNAL_SOURCE_EXISTS="test -f $srcdir/modules/afamqp/rabbitmq-c/librabbitmq/amqp.h"
-
-if $LIBRABBITMQ_INTERNAL_SOURCE_EXISTS; then
-    PYTHON_VERSION=`python -V 2>&1 | cut -d ' ' -f 2`
-    PYTHON_MAJOR=`echo $PYTHON_VERSION | cut -d '.' -f 1`
-    PYTHON_MINOR=`echo $PYTHON_VERSION | cut -d '.' -f 2`
-    if test $PYTHON_MAJOR -eq 2 && test $PYTHON_MINOR -lt 5; then
-        AC_MSG_WARN([Was about to execute rabbitmq configure script, but that requires Python >= 2.5, which you don't seem to have, reverting to system rabbitmq])
-        with_librabbitmq_client="system"
-    else
-        AC_CONFIG_SUBDIRS([modules/afamqp/rabbitmq-c])
-        LIBRABBITMQ_SUBDIRS="modules/afamqp/rabbitmq-c"
-    fi
-fi
-
-if test "x$with_librabbitmq_client" = "xinternal"; then
-    if $LIBRABBITMQ_INTERNAL_SOURCE_EXISTS; then
-        # these can only be used in modules/amqp as it assumes
-        # the current directory just one below rabbitmq-c
-
-        LIBRABBITMQ_LIBS="-L\$(top_builddir)/modules/afamqp/rabbitmq-c/librabbitmq -lrabbitmq"
-        LIBRABBITMQ_CFLAGS="-I\$(top_srcdir)/modules/afamqp/rabbitmq-c/librabbitmq -I\$(top_builddir)/modules/afamqp/rabbitmq-c/librabbitmq"
-    else
-        AC_MSG_WARN([Internal librabbitmq-client sources not found in modules/afamqp/rabbitmq-c])
-        with_librabbitmq_client="no"
-    fi
-elif test "x$with_librabbitmq_client" = "xsystem"; then
-    PKG_CHECK_MODULES(LIBRABBITMQ, librabbitmq >= $LIBRABBITMQ_MIN_VERSION,, enable_amqp="no")
-fi
-
-if test "x$with_librabbitmq_client" = "xno"; then
```

Comment:
Similar validation still needed: if amqp is enabled, librabbitmq-client must be available.